gcloud/storage/connection.py

```diff
@@ -256,7 +256,7 @@ def api_request(self, method, path, query_params=None,
             content_type = response.get('content-type', '')
             if not content_type.startswith('application/json'):
                 raise TypeError('Expected JSON, got %s' % content_type)
-            return json.loads(content)
+            return json.loads(content.decode('utf-8'))
```

Comment:
It is a conceptual error to "decode" unicode with a bytes->unicode encoding.  Please add the check.